src/share/classes/com/sun/javatest/regtest/agent/JUnitRunner.java

```diff
@@ -170,4 +178,62 @@ private static void runWithJUnitPlatform(Class<?> mainClass) throws Exception {
             throw new Exception(JUNIT_NO_DRIVER, ex);
         }
     }
+
+    static class PrintingListener implements TestExecutionListener {
+
+        final PrintWriter printer;
+        final Lock lock;
+
+        PrintingListener(PrintStream stream) {
+            this(new PrintWriter(stream, true));
+        }
+
+        PrintingListener(PrintWriter printer) {
+            this.printer = printer;
+            this.lock = new ReentrantLock();
+        }
+
+        @Override
+        public void executionFinished(TestIdentifier identifier, TestExecutionResult result) {
+            if (identifier.isTest()) {
+                lock.lock();
+                try {
+                    // always print the status of a test
+                    printer.println(identifier.getDisplayName() + " -> " + result.getStatus());
+                    // additionally print stacktrace for a non-successful test
+                    if (result.getStatus() != TestExecutionResult.Status.SUCCESSFUL) {
+                        result.getThrowable().ifPresent(printer::println);
+                    }
+                }
+                finally {
+                    lock.unlock();
+                }
+            }
+        }
+
+        @Override
+        public void executionSkipped(TestIdentifier identifier, String reason) {
+            if (identifier.isTest()) {
+                lock.lock();
+                try {
+                    printer.println(identifier.getDisplayName() + " -> SKIPPED because of: " + reason);
+                }
+                finally {
+                    lock.unlock();
+                }
+            }
+        }
+
+        @Override
+        public void reportingEntryPublished(TestIdentifier identifier, ReportEntry entry) {
```

Comment:
@AlanBateman Here is the place where a report entry is printed in the context of a test method.